internal/resources/cluster/resource_cluster.go

```diff
@@ -295,42 +295,63 @@ func resourceClusterDelete(_ context.Context, d *schema.ResourceData, m interfac
+	if d.HasChange(helper.GetFirstElementOf(SpecKey, tkgServiceVsphereKey, distributionKey, versionKey)) {
+		newVersion := d.Get(helper.GetFirstElementOf(SpecKey, tkgServiceVsphereKey, distributionKey, versionKey))
+		cluster.Spec.TkgServiceVsphere.Distribution.Version = newVersion.(string)
 	}
+}
 
-	getResp, err := config.TMCConnection.ClusterResourceService.ManageV1alpha1ClusterResourceServiceGet(constructFullname(d))
-	if err != nil {
-		return diag.FromErr(errors.Wrapf(err, "Unable to get Tanzu Mission Control cluster entry, name : %s", d.Get(clusterNameKey)))
+func withClusterGroupUpdate(_ authctx.TanzuContext, d *schema.ResourceData, cluster *clustermodel.VmwareTanzuManageV1alpha1ClusterCluster) {
+	if d.HasChange(helper.GetFirstElementOf(SpecKey, clusterGroupKey)) {
+		newClusterGroupName := d.Get(helper.GetFirstElementOf(SpecKey, clusterGroupKey))
+		if newClusterGroupName.(string) != "" {
+			cluster.Spec.ClusterGroupName = newClusterGroupName.(string)
+		}
 	}
+}
 
+func withMetaUpdate(config authctx.TanzuContext, d *schema.ResourceData, cluster *clustermodel.VmwareTanzuManageV1alpha1ClusterCluster) {
 	if common.HasMetaChanged(d) {
-		meta := common.ConstructMeta(d)
+		getResp, err := config.TMCConnection.ClusterResourceService.ManageV1alpha1ClusterResourceServiceGet(constructFullname(d))
+		if err != nil {
+			diag.FromErr(errors.Wrapf(err, "Unable to get Tanzu Mission Control cluster entry, name : %s", d.Get(clusterNameKey)))
+			return
+		}
+
+		objectMeta := common.ConstructMeta(d)
 
 		if value, ok := getResp.Cluster.Meta.Labels[common.CreatorLabelKey]; ok {
-			meta.Labels[common.CreatorLabelKey] = value
+			objectMeta.Labels[common.CreatorLabelKey] = value
 		}
 
-		getResp.Cluster.Meta.Labels = meta.Labels
-		getResp.Cluster.Meta.Description = meta.Description
+		cluster.Meta.Labels = objectMeta.Labels
+		cluster.Meta.Description = objectMeta.Description
+	}
+}
+
+func resourceClusterInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	// Get call to initialise the cluster struct
+	getResp, err := config.TMCConnection.ClusterResourceService.ManageV1alpha1ClusterResourceServiceGet(constructFullname(d))
+	if err != nil {
+		return diag.FromErr(errors.Wrapf(err, "Unable to get Tanzu Mission Control cluster entry, name : %s", d.Get(clusterNameKey)))
 	}
 
-	incomingCGName := d.Get(helper.GetFirstElementOf(SpecKey, clusterGroupKey))
+	var updates []func(authctx.TanzuContext, *schema.ResourceData, *clustermodel.VmwareTanzuManageV1alpha1ClusterCluster)
+
+	updates = append(updates, withMetaUpdate, withClusterGroupUpdate, withTKGsVsphereVersionUpdate, withTKGmVsphereVersionUpdate)
 
-	if incomingCGName.(string) != "" {
-		getResp.Cluster.Spec.ClusterGroupName = incomingCGName.(string)
+	for _, update := range updates {
+		update(config, d, getResp.Cluster)
```

Comment:
enhance the update function to return a boolean if there an update, 
and if there the boolean is set and then invoke the update function

Reply:
Return bool by every function seems redundant, making use of a var instead